tools/swift-machine-learning-api/codes/trainer/input_pipeline_dask.py

```diff
@@ -0,0 +1,521 @@
+
+"""
+Input Pipeline for loading data into the model
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from multiprocessing import cpu_count
+
+from google.cloud import storage
+import numpy as np
+import dask.dataframe as dd
+import dask
+import tensorflow as tf
+import lime.lime_tabular
+import dill
+import pandas as pd
+import os
+
+
+class InputReader(object):
+    """Class for reading input from different sources
+    Assuming csv files for now
+    """
+
+    def __init__(self, csv_path, task_type, target_var, na_values=None,
+                 column_names=None,
+                 to_drop=None, gcs_path=False, data_type=None):
+        """The init method initialise and keeps track of the source of input
+        (say csv, json etc) and other variables.
+
+        Arguments:
+                csv_path : string, Path of the csv files whether local or on remote storage
+                task_type : string, ML task at hand, following options are expected
+                        [classification, regression, clustering]
+                target_var : string, Name of the dependent/target variable
+                na_values : string, String by which the na values are represented in the data
+                column_names : string, Names of the columns passed in a text file
+                to_drop : list, Any redundant columns which can be dropped
+                gcs_path : boolean, Whether the csv is stored on google cloud storage
+                data_type : dict, dictionary containing the data type of all columns in format
+                    {'a': 'float', 'b': 'object', 'c': 'int' }
+        """
+        self.csv_path = csv_path
+        self.task_type = task_type
+        self.target_var = target_var
+        self.na_values = na_values
+        self.to_drop = to_drop
+        self.gcs_path = gcs_path
+        self.data_type = data_type
+        if column_names:
+            with tf.gfile.Open(column_names, 'r') as f:
+                self.column_names = [line.rstrip() for line in f]
+                self.column_names = [
+                    line for line in self.column_names if line]
+        else:
+            self.column_names = column_names
+
+    def parse_csv_wrap(self):
+        """
+        A Wrapper function for parsing csv files
+
+        Returns: 
+            _parse_csv function
+        """
+        return self._parse_csv()
+
+    def _parse_csv(self):
+        """Reads csv files in dask to determine the datatypes and other features about data
+        this helps in creating a dataset object in tensorflow
+
+        Returns:
+                df : dask dataframe, parsed dataframe object
+                list(df.columns) : list, list of column names
+        """
+        if self.gcs_path:
+            if isinstance(self.csv_path, list):
+                for index, path in enumerate(self.csv_path):
+                    bucket = path.split('//')[1].split('/')[0]
```

Comment:
refactor this to use regex matching, it is much more readable.

Reply:
Modified to use urlparse.